jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -44,6 +45,13 @@
       throws IOException {
     List<TarArchiveEntry> tarArchiveEntries = new ArrayList<>();
 
+    // Adds the extraction path itself; to do this, we are adding the current directory to act as
+    // the file input (since all directories are treated the same in TarArchiveEntry).
+    TarArchiveEntry extractionPathDirectoryEntry =
+        new TarArchiveEntry(Paths.get(".").toFile(), layerEntry.getExtractionPath());
```

Comment:
Given https://github.com/GoogleContainerTools/jib/issues/727#issuecomment-409951959, I guess the extraction path is like `/app/classes`, this directory gets the readable permission, but `/app` does not.

Reply:
Hmm, yea that might be it. Will try to think of a good way to add parent directories of the extraction too.